src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -379,7 +376,8 @@ public unsafe void SendResponseHeaders(bool appCompleted)
             fixed (byte* pReasonPhrase = reasonPhraseBytes)
             {
                 // This copies data into the underlying buffer
-                NativeMethods.http_set_response_status_code(_pHttpContext, (ushort)StatusCode, pReasonPhrase);
+                var hr = NativeMethods.http_set_response_status_code(_pHttpContext, (ushort)StatusCode, 0, pReasonPhrase);
+                CheckHResult(hr);
```

Comment:
What happens if this throws?

Reply:
We should throw I think as this is a fundamental failure. Investigating where the control flow will go.

Reply:
It would propagate up to whoever called read or write (to user code). Is that okay?

Reply:
You'd want to control/wrap the exception type in that case. Stream should only throw IOException and InvalidOperationExeption for most scenarios.

Reply:
Ahhh that makes sense. Would that be done in the stream itself? I will need to verify which exceptions are being thrown via bad hr status codes.

Reply:
Yes, somewhere up the stack between here and your response stream you'll need to catch this exception and wrap it.

On a related note, HttpSys and Kestrel's response streams do not throw exceptions for most failure scenarios. The caller usually doesn't care and it makes a lot of log noise. You may end up catching this exception at the stream level and suppressing it.

Reply:
So would this be a good place to catch the exceptions:
https://github.com/aspnet/IISIntegration/blob/dev/src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs#L40
and 
https://github.com/aspnet/IISIntegration/blob/dev/src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpResponseBody.cs#L57?
Or would it be better to catch it after the managed to native write.

Reply:
More like here:
https://github.com/aspnet/IISIntegration/blob/94c895a1bd1b9bba48ce496200fd386322fb6a74/src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs#L249
And this only applies to the response body. You don't call this from the request body.